server.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"sync"
+	"time"
+)
+
+// Server is an interface implemented by a value that can provide values to a
+// handler in a separate goroutine.
+type Server interface {
+	// Close prevents the server from sending any more data to a handler.
+	// A call to this method will block until the server has finished
+	// processing all existing data.
+	// This method will fail if the server is already closed.
+	Close() error
+}
+
+// A UnixServer represents a listener on a unix socket that forwards data to a
+// handler.
+// This server expects data sent over a connection to be terminated with a '\n'
+// symbol.
+// The result from the handler will be sent back across the TCP connection,
+// with each entry (assignment and location) terminated with a '\n' symbol.
+type UnixServer struct {
+	listener  *net.UnixListener
+	waitGroup *sync.WaitGroup
+	schan     chan time.Time
+}
+
+// TODO: Comply with Close()
```

Comment:
question for whoever knows: do we do TODO(name): instead of TODO: outside of the website?